### PR TITLE
Add support to TXT records

### PR DESCRIPTION
**Pull request by dth0**
This PR adds support to TXT record type to xk6-dns.

I am just extending the existing logic to support TXT entries alongside the other record types. The implementation follows the same patterns already in place, so it should feel pretty consistent with the rest of the codebase.

Tested with a few sample records to make sure everything parses and behaves as expected.

Let me know if you’d like me to add more test cases or cover additional edge cases

### CODE REVIEW

File: dns/nameserver.go [hunk 1]
```diff
@@ -26,19 +26,38 @@ func (n Nameserver) Addr() string {
...
 	ip := net.ParseIP(hostStr)
+	if ip != nil {
+		return Nameserver{ip, port}, nil
+	}
+
+	// Not an IP, try resolving the hostname
+	ips, err := net.LookupHost(hostStr)
+	if err != nil {
+		return Nameserver{}, fmt.Errorf("failed to resolve nameserver hostname %s: %w", hostStr, err)
+	}
```
Comment (oleiade) [inline dns/nameserver.go]: `net.LookupHost` here will use the Go stdlib network dialer. In k6 that would turn out to be a problem as k6 exposes its own dialer in order to keep track of what is going on network-wise.

The extension's client holds the k6 dialer, and whatever we do that involves networking should be done through the k6 dialer 🙇

File: dns/client.go [hunk 2]
```diff
@@ -116,13 +117,15 @@ func (r *Client) Resolve(
...
 	for _, a := range response.Answer {
 		switch t := a.(type) {
 		case *dns.A:
-			ips = append(ips, t.A.String())
+			results = append(results, t.A.String())
 		case *dns.AAAA:
-			ips = append(ips, t.AAAA.String())
+			results = append(results, t.AAAA.String())
+		case *dns.TXT:
+			results = append(results, t.Txt...)
```
Comment (oleiade) [inline dns/client.go]: One TXT-specific DNS detail we should handle here: `dns.TXT.Txt` is a slice of character-strings within a single TXT RR, not a slice of separate TXT records. 

Large TXT records are commonly split into multiple chunks on the wire, so append(results, t.Txt...) can expose one DNS record as multiple JS results.

I think we should consider preserving one array element per TXT RR:
```
case *dns.TXT:
    results = append(results, strings.Join(t.Txt, ""))
```

**Review by oleiade — CHANGES_REQUESTED** [review]
Apologies for the very slow response time, this PR got buried in a massive pile of Renovate PRs.

Great work, but two blockin comments on my end I'd like to see addressed before merging. Let me know if you need anything from me to help with the PR 🙇🏻

File: dns/nameserver.go (hunk 1 above)
Comment (dth0) [reply dns/nameserver.go]: Hey man, sorry for the late reply.

Sorry, I didn't pay attention to that. I moved it to the outermost function (the only one that uses it) so I can have access to the internal dialer.

Please let me know if this makes sense now.

File: dns/client.go (hunk 2 above)
Comment (dth0) [reply dns/client.go]: that makes sense, just push a commit with your suggestion.
lmk wdyt

File: dns/nameserver.go (hunk 1 above)
Comment (dth0) [reply dns/nameserver.go]: Just added a quick fix to better handle entries in the form of "HOST:PORT".

File: dns/module.go [hunk 3]
```diff
@@ -97,6 +98,31 @@ func (mi *ModuleInstance) Resolve(query, recordType, nameserverAddr sobek.Value)
...
+
+		if len(ipAddrs) == 0 {
+			reject(fmt.Errorf("nameserver hostname %s resolved to no addresses", host))
+			return promise
+		}
+
+		// Replace hostname with resolved IP, preserving original port
+		nameserverAddrStr = ipAddrs[0] + nameserverAddrStr[len(host):]
+	}
+
```
Comment (oleiade) [inline dns/module.go]: Thanks for the iterations here. The TXT support itself looks good now: `strings.Join(t.Txt, "")` (one element per RR) and the EDNS0 buffer are good to go, and the tests cover both the existing and NXDOMAIN cases. Both of my earlier blocking comments are addressed.

One bigger-picture request before we merge: Let's scope this PR down to TXT.

The nameserver hostname-resolution block in `Resolve()` (the `parseHostAndPort` + `mi.dnsClient.Lookup` + string rebuild) feels like a separate feature from TXT records and I believeTXT support needs none of it?

I'd prefer to keep this PR focused on TXT and move the hostname-nameserver capability (as I understand it) to its own PR if my understanding is correct (that's it's not mandator) and relevant.

With this done, I'm happy to merge this PR 🙇

**Review by oleiade — CHANGES_REQUESTED** [review]

File: dns/module.go (hunk 3 above)
Comment (dth0) [reply dns/module.go]: I agree, this feature has not to do with the TXT, for now I will drop it to now block this PR and later working in another one specific to this case

**Review by oleiade — APPROVED** [review]
Thanks for the changes 🙇🏻 

This looks good to merge from my perspective, can we get your eyes on this @ankur22 @szkiba ?

**Review by ankur22 — APPROVED** [review]
LGTM 🚀 Thanks for the contribution @dth0!

**Review by szkiba — APPROVED** [review]
